lib/timers.js

```diff
@@ -229,7 +234,7 @@ function tryOnTimeout(timer, list) {
   timer._called = true;
   var threw = true;
   try {
-    timer._onTimeout();
+    ontimeout(timer);
```

Comment:
Which function? `tryOnTimeout()`? `ontimeout()`? The user's callback?

I don't know that throwing causes a deopt or not, but adding a `try-*` block certainly does cause a deopt for the containing function.